test/util/util-test.ts

```diff
@@ -24,3 +25,87 @@ describe('typeIsArray', () => {
     typeIsArray(null).should.be.false();
   });
 });
+
+describe('asyncMergeSort', () => {
+  const sortAsc = async (a: number | string, b: number | string) => {
+    if (a < b) return -1;
+    else if (a > b) return 1;
+    return 0;
+  };
+  const sortDesc = async (a: number | string, b: number | string) => {
+    if (a < b) return 1;
+    else if (a > b) return -1;
+    return 0;
+  };
+
+  it('should not change empty array', async () => {
+    const arr: any = [];
+    const sorted = await asyncMergeSort(arr, sortAsc);
+
+    sorted.should.eql([]);
+  });
+
+  it('should not change array with 1 element', async () => {
+    const arr = [1];
+    const sorted = await asyncMergeSort(arr, sortAsc);
+
+    sorted.should.eql([1]);
+  });
+
+  it('should sort array with 2 elements ascending', async () => {
+    const arr = [2, 1];
+    const sorted = await asyncMergeSort(arr, sortAsc);
+
+    sorted.should.eql([1, 2]);
+  });
+
+  it('should sort array with 2 elements descending', async () => {
+    const arr = [1, 2];
+    const sorted = await asyncMergeSort(arr, sortDesc);
+
+    sorted.should.eql([2, 1]);
+  });
+
+  it('should sort array ascending', async () => {
+    const arr = [3, -1, 0, 1, 5, 2];
+    const sorted = await asyncMergeSort(arr, sortAsc);
+
+    sorted.should.eql([-1, 0, 1, 2, 3, 5]);
+  });
+
+  it('should sort array descending', async () => {
+    const arr = [3, -1, 0, 1, 5, 2];
+    const sorted = await asyncMergeSort(arr, sortDesc);
+
+    sorted.should.eql([5, 3, 2, 1, 0, -1]);
+  });
+
+  it('should sort strings ascending', async () => {
+    const arr = ['def', 'abc', 'jkl', 'ghi'];
+    const sorted = await asyncMergeSort(arr, sortAsc);
+
+    sorted.should.eql(['abc', 'def', 'ghi', 'jkl']);
+  });
+
+  it('should sort complex object', async () => {
+    const arr = [{ val: 3 }, { val: 1 }, { val: 2 }];
+
+    const sorted = await asyncMergeSort(arr, async (a, b) => a.val - b.val);
+    sorted.should.eql([{ val: 1 }, { val: 2 }, { val: 3 }]);
+  });
+});
+
+describe('isPromise', () => {
+  it('should return false for nullish value', () => {
+    isPromise(undefined).should.be.false();
+    isPromise(null).should.be.false();
+  });
+
+  it('should return false for non-promise', () => {
+    isPromise('not a promise').should.be.false();
+  });
+
+  it('should return true for promise', () => {
+    isPromise(Promise.resolve('definitely a promise')).should.be.true();
+  });
+});
```

Comment:
If you kill `isPromise` you can obviously kill these tests too.

Reply:
RIP!